OCPBUGS-65484: Add debug-useful ClusterOperator relatedObjects

RadekManak opened the pull request:
Expand `machine-approver` ClusterOperator `relatedObjects` with RBAC that namespace inspect does not collect.

Keeps the existing namespace and CertificateSigningRequests entries.

## Added relatedObjects
- ClusterRoles / ClusterRoleBindings: `system:openshift:controller:machine-approver`, `system:openshift:controller:machine-approver-capi`
- Role / RoleBinding: `machine-approver` in `openshift-config-managed`
- Roles / RoleBindings in `openshift-cluster-machine-approver`

## Tests
Parity test keeping the Go list in sync with the ClusterOperator manifest.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved ClusterOperator status reporting by including all relevant machine approver RBAC resources.
  * Added references for controller roles, bindings, and namespace-scoped permissions.

* **Tests**
  * Added validation to keep reported related resources consistent between deployment configuration and runtime status.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

wking commented on status.go:
```diff
@@ -46,6 +46,11 @@ var relatedObjects = []osconfigv1.ObjectReference{
 	},
+	{
+		Group:    "rbac.authorization.k8s.io",
+		Resource: "clusterroles",
```
Is a ClusterRole all you need?  I'd expect you'd also need a ClusterRoleBinding that attaches that role to a ServiceAccount?  And maybe other things?

damdo commented:
/remove-lifecycle rotten

RadekManak committed:
OCPBUGS-65484: Add debug-useful ClusterOperator relatedObjects

Add ClusterRoles/ClusterRoleBindings, foreign-namespace config-managed
Role/RoleBinding, operator-namespace Roles/RoleBindings, CSRs, and
PrometheusRules to relatedObjects so inspect and must-gather collect the
resources needed to debug machine-approver.

Keep Go and the ClusterOperator manifest in sync with a parity test.

RadekManak commented on related_objects_test.go:
```diff
@@ -0,0 +1,24 @@
+	"sigs.k8s.io/yaml"
+)
+
+var _ = Describe("relatedObjects", func() {
+	It("should stay in sync with the ClusterOperator manifest", func() {
+		manifestPath := filepath.Join("manifests", "05-clusteroperator.yaml")
+		data, err := os.ReadFile(manifestPath)
+		Expect(err).ToNot(HaveOccurred(), "should be able to read ClusterOperator manifest")
+
+		co := &osconfigv1.ClusterOperator{}
+		Expect(yaml.Unmarshal(data, co)).To(Succeed(), "should be able to unmarshal ClusterOperator manifest")
+
+		Expect(co.Status.RelatedObjects).To(Equal(relatedObjects), "Go relatedObjects must match the static ClusterOperator manifest")
```
This would just result having a third place for the same list. The test is meant to catch people adding to one and not the other.

RadekManak commented:
/hold cancel

RadekManak commented:
/retest-required
